cdap-common/src/main/java/co/cask/cdap/common/service/ServiceDiscoverable.java

```diff
@@ -34,6 +36,15 @@ public static String getName(String namespaceId, String applicationId, String se
                          applicationId, serviceId);
   }
 
+  public static ProgramId getId(String name) {
+    String[] parts = name.split("\\.");
```

Comment:
How often does this method get called? If it is on every service request, it is better to use a `Pattern` to split the string, since the `String.split` methods involve compiling a new regex every time. Or even faster way is to use `indexOf('.')` since you know exactly how many parts are there, there is no need to use regex.